core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -146,8 +143,13 @@ public void init() throws StoreInitializationException {
     props.put(ProducerConfig.RETRIES_CONFIG, 0); // Producer should not retry
     producer = new KafkaProducer<byte[],byte[]>(props);
 
-    // start the background thread that subscribes to the Kafka topic and applies updates
-    kafkaTopicReader.start();
+    // start the background thread that subscribes to the Kafka topic and applies updates.
+    // the thread must be created after the schema topic has been created.
+    this.kafkaTopicReader =
+            new KafkaStoreReaderThread<>(this.bootstrapBrokers, topic, groupId,
+                    Integer.MIN_VALUE, this.storeUpdateHandler,
```

Comment:
Not part of this patch, but I'm kind of puzzled with commit interval passed here. Doesn't this effectively turn off commits? If so, then do we even need the commit logic implemented in KafkaStoreReaderThread?